phoenix-core/src/main/java/org/apache/phoenix/coprocessor/UngroupedAggregateRegionObserver.java

```diff
@@ -1103,13 +1104,21 @@ private RegionScanner rebuildIndices(final RegionScanner innerScanner, final Reg
             if (oldCoproc) {
                 return new IndexerRegionScanner(scanner, region, scan, env, this);
             } else {
-                return new IndexRebuildRegionScanner(scanner, region, scan, env, this);
+                if (region.getTableDesc().hasCoprocessor(IndexRegionObserver.class.getCanonicalName())) {
```

Comment:
Don't we have IndexRegionObserver for new design all the time? What is the case when the index doesn't have both Indexer and IndexRegionObserver?

Reply:
@gokceni IndexRegionObserver is only on the data table. Index table has neither Indexer nor IndexRegionObserver

Reply:
Yes you are right @tkhurana. I meant data table. Don't we remove Indexer and add IndexRegionObserver during upgrade?

Reply:
Is this an optimization for checking if this table is Index table @kadirozde rather than querying the PTable and looking at its type?

Reply:
We have not removed Indexer yet. Regarding the PTable comment, this is a server side code and in general we do not want to access the syscat on the server side for performance reasons mainly.